### ISSUE TITLE
[BITV] 9.1.4.3/7.1 - The label of the "Change color" button in the "Background" section does not meet the contrast requirements for text - it is 4,16:1. (1)

### ISSUE THREAD

**Issue by JuliaKirschenheuter** [nextcloud/server#41849]
https://report.bitvtest.de/default-en/cbedb5ef-4815-4cba-a83d-995599c178c3.html#checkpoint-6fe46b78da-v7-n1

**Comment by ShGKme** [comment]
There is no pair of text colors that is contrast enough on any background. E.g. in some gray of blue backgrounds, both white and black text is not contrast enough.

An alternative could be to add a dark transparent gradient, but it would break the selected color.

My current idea:

![Image](https://github.com/nextcloud/server/assets/25978914/9c75252a-4378-4657-af9d-e8390170d6fc)

<details>

```css
    background: var(--color-main-background);
    padding: 0 calc(var(--default-grid-baseline) * 4);
    border-radius: var(--border-radius-rounded);
    display: inline-flex;
    height: var(--default-clickable-area);
    justify-content: center;
    align-items: center;
```

</details>

@nextcloud/designers what do you think about this proposal?

**Comment by ShGKme** [comment]
Another alternative could be to separate **Background** and **Primary color** settings, and do not use color as a background of a UI element at all:


![Image](https://github.com/nextcloud/server/assets/25978914/cae74c22-5cb1-4e6f-83d0-188b886b82cb)

**Comment by szaimen** [comment]
> An alternative could be to add a dark transparent gradient, but it would break the selected color.
> 
> My current idea:
> 
> ![Image](https://private-user-images.githubusercontent.com/25978914/289624381-9c75252a-4378-4657-af9d-e8390170d6fc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQxOTIyMDEsIm5iZiI6MTcwNDE5MTkwMSwicGF0aCI6Ii8yNTk3ODkxNC8yODk2MjQzODEtOWM3NTI1MmEtNDM3OC00NjU3LWFmOWQtZTgzOTAxNzBkNmZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAyVDEwMzgyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjODI0MjM1OTRlYTIzM2VhMWY4YjRmZmZkNzUxNjgwMmMyOTY1NDhlZDE5ZWE3MTQ4MDc2ZGFjNjM1NzI1ODMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qRxaRaugZsEiI71RmVO-EAsAk2sAdMMZQAMQdEmr9SE)
> 
> @nextcloud/designers what do you think about this proposal?

The idea and mockup looks good to me :)

**Comment by ShGKme** [comment]
Update: there is exactly the same button on the admin settings.

**Comment by ShGKme** [comment]
Backport required

**Comment by JuliaKirschenheuter** [comment]
closing as done
